File: src/FoodOrderSystem.Persistence.MongoDB/OrderModel.cs
using System;
using FoodOrderSystem.Domain.ViewModels;

namespace FoodOrderSystem.Persistence.MongoDB
{
    public class OrderModel
    {
        public Guid Id { get; set; }
        
        public CustomerInfoModel CustomerInfo { get; set; }
        
        public CartInfoModel CartInfo { get; set; }
        
        public string Comments { get; set; }
        
        public Guid PaymentMethodId { get; set; }
        
        public double Costs { get; set; }
        
        public DateTime CreatedOn { get; set; }
        
        public DateTime? UpdatedOn { get; set; }
        
        public Guid? UpdatedBy { get; set; }
    }
}

File: src/Gastromio.App/ClientApp/src/app/order/models/order.model.ts
import { PaymentMethodModel } from '../../shared/models/payment-method.model';

import { CustomerInfoModel } from './customer-info.model';
import { CartInfoModel } from './cart-info.model';

export class OrderModel {
  constructor(init?: Partial<OrderModel>) {
    if (init) {
      Object.assign(this, init);
    }
    this.customerInfo = this.customerInfo
      ? new CustomerInfoModel(this.customerInfo)
      : undefined;
    this.cartInfo = this.cartInfo
      ? new CartInfoModel(this.cartInfo)
      : undefined;
    this.paymentMethod = this.paymentMethod
      ? new PaymentMethodModel(this.paymentMethod)
      : undefined;
  }

  public id: string;

  public customerInfo: CustomerInfoModel;

  public cartInfo: CartInfoModel;

  public comments: string;

  public paymentMethodId: string;

  public paymentMethod: PaymentMethodModel;

  public valueOfOrder: number;

  public costs: number;

  public totalPrice: number;

  public clone(): OrderModel {
    return new OrderModel({
      id: this.id,
      customerInfo: this.customerInfo?.clone(),
      cartInfo: this.cartInfo?.clone(),
      comments: this.comments,
      paymentMethodId: this.paymentMethodId,
      paymentMethod: this.paymentMethod?.clone(),
      valueOfOrder: this.valueOfOrder,
      costs: this.costs,
      totalPrice: this.totalPrice,
    });
  }
}
